cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -113,6 +113,7 @@ var (
 	AvailableExpanders = []string{RandomExpanderName, MostPodsExpanderName, LeastWasteExpanderName}
 	expanderFlag       = flag.String("expander", RandomExpanderName,
 		"Type of node group expander to be used in scale up. Available values: ["+strings.Join(AvailableExpanders, ",")+"]")
+	autoDiscoverNodeGroup = flag.Bool("auto-discover-node-group", false, "Enables auto discovery of node groups. Currently supported only in AWS.")
```

Comment:
We should fail fast or a log some sort of warning/error if someone has this flag enabled but not on AWS.